### PR TITLE
fix sync problem during reblance

### PR DESCRIPTION
**Pull request by weiliu1031**
Signed-off-by: Wei Liu <wei.liu@zilliz.com>
issue: https://github.com/milvus-io/milvus/issues/18178

### CODE REVIEW

File: internal/querycoord/task.go [hunk 1]
```diff
@@ -2485,11 +2472,7 @@ func (lbt *loadBalanceTask) globalPostExecute(ctx context.Context) error {
 		}
 	}
 
-	err := wg.Wait()
```
Comment (yah01) [inline internal/querycoord/task.go:2488]: Have to wait here to make sure the shard leader is updated before syncReplicaSegments()

**Review by yah01 — CHANGES_REQUESTED** [review]

File: internal/querycoord/task.go (hunk 1 above)
Comment (weiliu1031) [reply internal/querycoord/task.go:2488]: fixed!

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**